test/integration/enterprise_test.go

```diff
@@ -67,77 +73,100 @@ func TestKongEnterprisePostgres(t *testing.T) {
 	require.NotNil(t, adminURL)
 
 	t.Log("building a GET request to gather admin api information")
-	req, err := http.NewRequestWithContext(ctx, "GET", adminURL.String(), nil)
+	req, err := http.NewRequestWithContext(ctx, http.MethodGet, adminURL.String(), nil)
 	require.NoError(t, err)
-	req.Header.Set("Kong-Admin-Token", adminPassword)
-
-	t.Run("verifying the admin api version is enterprise", func(t *testing.T) {
-		t.Log("pulling the admin api information")
-		httpc := http.Client{Timeout: time.Second * 10}
-		var body []byte
-		require.Eventually(t, func() bool {
-			resp, err := httpc.Do(req)
-			if err != nil {
+	if adminPassword != "" {
+		decorateRequestWithAdminPassword(t, req, adminPassword)
+	} else {
+		t.Log("skipping setting the admin api password (Kong-Admin-Token header)")
+	}
+	t.Log("verifying the admin api version is enterprise")
+	httpClient := &http.Client{Timeout: time.Second * 10}
+	test.EventuallyExpectedStatusCodeAndBody(t, httpClient, req, http.StatusOK, test.WithBodyChecker(
+		func(t *testing.T, body []byte) bool {
```

Comment:
This whole check / option I believe could be made into a named option like e.g. `WithBodyHasKongEnterpriseVersion`. WDYT?

Reply:
But you need a combination of the right request and function for checking, here we explicitly pass a request

Reply:
That `WithBodyHasKongEnterpriseVersion()` will work only when `req` will contain a valid URL and method. Another approach is to check it based on the header returned by Gateway, but idk if the format is the same as in current approach

Reply:
I don't mind using body or response header: whatever works. I'm just concerned about the interface here.

I don't see a problem with the request not containing a proper URL or a method. It will just failed on `http.Do()` call right? Before the option kicks in.